File: cpp/barrelsize.cpp
#include <iostream>
#include <cmath>

int main()
{
    //Declaring variables and constants
    const double pi = 3.14159265358979323846; //const pi
    double radius   = 0.0;
    double height   = 0.0;
    double volume   = 0.0;
    double area     = 0.0;
    double liters   = 0.0;
    double floz     = 0.0;
    double gallons  = 0.0;

    //take user input (radius, height)
    std::cout << "Enter radius of the barrel base in milimeters: ";
    std::cin >> radius;
    std::cout << "\nEnter height of the barrel in milimeters: ";
    std::cin >> height;

    //calculate results
    area = 2 * pi * (radius + height);
    volume = pi * pow(radius, 2) * height;
    liters = volume / 1000.0;
    floz = liters * 0.264172;
    gallons = floz / 128.0;

   //print results
   std::cout << "The barrel base radius is: " << radius << " mm.\n";
   std::cout << "The barrel height is: " << height << " mm.\n";
   std::cout << "The surface area of the barrel is: " << area << " mm^2.\n";
   std::cout << "The volume of the barrel is: " << volume << " mm^3.\n";
   std::cout << "The barrel can hold " << liters << " liters,\n";
   std::cout << "which is " << floz << " fluid ounces\n";
   std::cout << "or " << gallons << " gallons." << std::endl;

   return 0;
}

File: swift/barrelsize.swift
//Swift Version of the Barrelsize exercise
import Foundation

enum Const
{
    static let pi : Double = Double.pi
}

func calcVolume(h: Double, r: Double) -> (volume: Double, liters: Double, floz: Double, gal: Double) 
{
    let volume: Double = Const.pi * pow(r, 2) * h
    let liters: Double = volume / 1000
    let floz: Double = liters * 33.814
    let gal: Double = floz / 128.0
    return (volume, liters, floz, gal)
}

func calcArea(h: Double, r: Double) -> Double
{
    let area : Double = 2 * Const.pi * r * (r + h)
    return area
}

func main()
{
    print("Enter a height value in centimeters: ")
    let height = readLine()
    guard let heightUnwrapped = height, let h = Double(heightUnwrapped) else {
        print("Invalid height input, defaulting to 0")
        return
    }
    print("Enter a radius in centimeters: ")
    let radius = readLine()
    guard let radiusUnwrapped = radius, let r = Double(radiusUnwrapped) else {
        print("Invalid radius input, defaulting to 0")
        return
    }

    let results = calcVolume(h: h, r: r)
    print("Volume in ccm is: \(results.volume) ccm")
    print("Volume in Liters is: \(results.liters) L")
    print("Volume in Fluid Ounces is: \(results.floz) floz")
    print("Volume in Gallons is: \(results.gal) gal")
    print("Barrel area is \(calcArea(h: h, r: r)) sqcm")
}
main()